hack/docker/gen/main.go

```diff
@@ -52,7 +158,55 @@ var tmpl = fmt.Sprintf(`# syntax = docker/dockerfile:latest
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
-#
+#`
+)
+
+var license = template.Must(template.New("license").Parse(header + `
+
+# DO_NOT_EDIT this workflow file is generated by https://github.com/vdaas/vald/blob/main/hack/docker/gen/main.go
```

Comment:
I feel like hack/gen makes it hard to know what is generated from the path, so you might want to think about the name a bit more, I feel like docker is fine in terms of docker image and the GitHub action to build it, or component, or I feel like it could also be component or something like that.